R/generate_simulated_data.R

```diff
@@ -186,8 +186,8 @@ generate_simulated_data <- function(r_in_weeks = # nolint
   )
 
   forecast_date <- date_df |>
-    dplyr::filter(t == ot + nt) |>
-    dplyr::pull(date)
+    dplyr::filter(.data$t == !!ot + !!nt) |>
+    dplyr::pull(.data$date)
```

Comment:
Not for now, but for future reference this can be `dplyr::pull("date")`

Reply:
Can you explain why it should be here -- is it because it is a column name and we should always refer to column names by "col_name" if they are not variables for column names?

Reply:
"Should" might be strong, but the reason it _can_ goes back to the distinction between "tidy selection" and "data masking". `dplyr::select()` and `dplyr::pull()`, among others, use[^1] tidy selection; functions like `dplyr::mutate()` and `dplyr::summarise()` use[^1] data-masking.

[^1]: For positional arguments after the first argument, which is typically the data frame to be manipulated

See https://dplyr.tidyverse.org/articles/programming.html#introduction

Reply:
The result is that 

```R
df |> dplyr::select("a_string")
```
means "select the column named `'a_string'` if it exists"

whereas:

```R
df |> dplyr::mutate(new_col = "a_string")
```
is interpreted as "create a new column named `'new_col'` whose value in each row is the string `'a_string'`. 

And if what you want is "create a new column named `'new_col'` whose values are the values of another column named `'a_string'`, you need:

```R
df |> dplyr::mutate(new_col = .data[["a_string"]])
```

or 

```R
df |> dplyr::mutate(new_col = .data$a_string)
```

Reply:
Or:
```r
df |> dplyr::mutate("new_col" = .data$a_string
```
Which I believe is the syntax we're rolling with here correct (we should always explicitly use the `""` for creating a new column name)?